common/internal_model/src/main/scala/uk/ac/wellcome/models/work/internal/WorkType.scala

```diff
@@ -1,5 +1,160 @@
 package uk.ac.wellcome.models.work.internal
 
-case class WorkType(id: String,
-                    label: String,
-                    ontologyType: String = "WorkType")
+import enumeratum.{Enum, EnumEntry}
+import io.circe.{Decoder, Encoder, Json}
+
+sealed trait WorkType extends EnumEntry {
+  val id: String
+  val label: String
+}
+
+object WorkType extends Enum[WorkType] {
+  val values = findValues
+
+  implicit val workTypeEncoder: Encoder[WorkType] = Encoder.instance[WorkType] {
+    workType =>
+      Json.obj(
+        ("id", Json.fromString(workType.id)),
+        ("label", Json.fromString(workType.label))
+      )
+  }
+
+  implicit val workTypeDecoder: Decoder[WorkType] = Decoder.instance[WorkType] {
+    cursor =>
+      for {
+        id <- cursor.downField("id").as[String]
+      } yield {
+        fromCode(id).getOrElse(throw new Exception(s"Invalid WorkType id: $id"))
+      }
+  }
+
+  def fromCode(id: String): Option[WorkType] = {
+    values.find(workType => workType.id == id)
+  }
+
+  sealed trait UnlinkedWorkType extends WorkType
+  sealed trait LinkedWorkType extends WorkType {
+    val linksTo: UnlinkedWorkType
+  }
+
+  case object BooksWorkType extends UnlinkedWorkType {
```

Comment:
As this is already within the `WorkType` object (and thus already namespaced), in my opinion it is easier to read if we don't include "WorkType" again in the name. e.g. 

`WorkType.Books` vs `WorkType.BooksWorkType`